2016-06-28-Rust-Reach-Fractal.md

```diff
@@ -0,0 +1,91 @@
+
+### At the crossroads
+
+Now that we've shipped Tessel 2, there are several things we could do:
+
+1. Keep adding features in software, like support for more languages or additional features
+1. Build more modules to expand the platform
+1. Work on a "Tessel 3" that is an evolution of the board
+1. Work on a "Tessel X" that uses some of the same tooling but suits a different niche
+1. Something else entirely, which approaches the mission in a different way
+
+This is where the Steering Committee comes in. It's our job, as leaders of the project, to determine the vision and create a plan for future development.
+
+For us, this is the first major vision-type decision we've had to make since moving to an all-volunteer open governance model. It's challenging to craft cohesive vision in our limited face-to-face meeting time.
+
+To make a decision of this type, we need to lean back on our mission and core principles, on the logistical implications of our decisions, and on the needs of our community.
+
+Critically, we need to pick projects that are well-scoped and highly focused, so that we can progress measurably with the time and person-power we have available. All of these projects are arguably worthwhile. Which is best?
+
+## Fulfilling the vision
+
+Coming back to the core, we want to build the OSHW movement through our platform. This problem fits the original vision for Tessel 2, to help people of different backgrounds create products quickly and easily.
+
+We've come some of the way towards making Tessel 2 the base for a product you can take to market. But we're not all the way there yet. Here are some of the issues we see:
+
+1. It's not easy for most of our users to take the tools we've created so far and go to market. If you don't come from an electrical engineering background, the next step might be insurmountably daunting. You might not know what things you can afford to remove from the design, or how much efficiency or cost they would save you.
+1. Connected device products tend to exist in a networked configuration, usually involving several low-cost, low-power nodes. This makes Tessel 2 not well suited to a lot of product applications.
+1. You can optimize to a point, but at the end of the day you are running JavaScript. This might not be as efficient or as innately reliable as you want your product to be.
+
+## A path forward: Rust, Reach, and Fractal
+
+Drawing on some exploratory work we did over a year ago, we think we can work on these problems such that their solutions build upon each other. Here's the idea:
+
+1. Work on improved **Rust** support. The Rust language is more efficient and explicit than JavaScript and should serve well in production. The code is overall pretty legible, and it's a welcoming community with good documentation. We've already begun tooling for this on the [tessel-rust](//github.com/tessel/tessel-rust) repo.
```

Comment:
I would suggest being more specific than "more efficient". Perhaps "executes code much faster and more robustly".